source/trainer.py

```diff
@@ -0,0 +1,147 @@
+# RUN 3 lines below in a seperate cell in Google Colab
+# !pip install transformers tokenizers wandb huggingface_hub
+# from huggingface_hub import notebook_login
+# notebook_login()
+
+import os
+from pathlib import Path
+from transformers import (
+    PreTrainedTokenizerFast,
+    DataCollatorForLanguageModeling,
+    GPT2Config,
+    GPT2LMHeadModel,
+    TrainingArguments,
+    Trainer,
+)
+from tokenizers import Tokenizer
+from tokenizers.models import WordLevel
+from tokenizers.pre_tokenizers import WhitespaceSplit
+from tokenizers.trainers import WordLevelTrainer
+from datetime import datetime
+import wandb
+
+# CONFIG:
+TRAIN_FROM_CHECKPOINT = None  # Example: checkpoint-80000
+EVAL_STEPS = 1000
+PER_DEVICE_TRAIN_BATCH_SIZE = 10
```

Comment:
the default value should be much lower. 10 sometimes broke collab even with a small dataset
PER_DEVICE_TRAIN_BATCH_SIZE = 2 or 4